storage/storage.go

```diff
@@ -442,6 +456,8 @@ type SignedURLOptions struct {
 	// Scheme determines the version of URL signing to use. Default is
 	// SigningSchemeV2.
 	Scheme SigningScheme
+
+	endpoint string
```

Comment:
Now I'm wondering if we should just make this exported to support users who are using the standalone func as well. I don't really see a big downside to doing so. WDYT?

Reply:
I think the main concern I'd have is that this field gets overwritten in the bucketHandle methods (so it can get set to option.withEndpoint without users needing to add that to the opts). Although, I guess we could check and only override this field if not provided in opts - how does that sound? That way this field is exported and takes precedence over everything else (emulator env var, WithEndpoint option...).